db/document.go

```diff
@@ -332,6 +333,21 @@ func (doc *Document) GetMetaMap(userXattrKey string) (map[string]interface{}, er
 	}, nil
 }
 
+func (doc *Document) SetCrc32cUserXattrHash(userXattrKey string) {
+	if len(doc.rawUserXattr) > 0 {
+		doc.SyncData.Crc32cUserXattr = base.Crc32cHashString(doc.rawUserXattr)
+		return
+	}
+
+	// If userXattrKey is still set and there is no rawUserXattr then the xattr has been removed from the doc and we can
+	// set to empty hash.
+	// If userXattr key is no longer set but the crc32 hash is still set then we can set it to empty hash. This likely
+	// means the feature was enabled but later disabled.
+	if len(doc.rawUserXattr) == 0 && userXattrKey != "" || userXattrKey == "" && doc.SyncData.Crc32cUserXattr != "" {
+		doc.SyncData.Crc32cUserXattr = ""
+	}
```

Comment:
This is getting a bit hard to follow now... Fairly sure it can be rewritten to minimise the conditionals.

```suggestion
	// Empty the hash if we've no longer got a value or a key defined.
	if len(doc.rawUserXattr) == 0 || userXattrKey == "" {
		doc.SyncData.Crc32cUserXattr = ""
		return
	}

	doc.SyncData.Crc32cUserXattr = base.Crc32cHashString(doc.rawUserXattr)
```

Reply:
I considered this but this will mean that even if the user doesn't have, or has never had user xattrs enabled then this property will be added. Not really an issue in terms of space or anything but could be misleading.

Reply:
Aren't both of those cases caught by the conditional setting empty string?

Reply:
Not sure what you mean? Empty userXattrKey means that the feature is disabled.

Reply:
Always set an empty string if the feature is disabled? Is that not what we want?

Reply:
Well like I say, if the feature has never been enabled it'd be weird to have that as a key in sync data imo.

Reply:
That field is a string type. It's always going to be initialized to `""` regardless.

Reply:
Ah. Good point. Okay then.